part2/tests/array_test.h

```diff
@@ -410,27 +357,27 @@ class ArrayTest : public Test_template {
             OK("ArrayTest.pushBackInt");
         }
 
-        void putReplacesObjectInt() {
+        void setReplacesObjectInt() {
             IntArray* arr = new IntArray();
 
             int i1 = 5;
             int i2 = 10;
 
-            arr->put(4, i1);
-            arr->put(4, i2);
+            arr->set(0, i1);
```

Comment:
I don’t think set should work here since the size is 0

Reply:
Fixed that and used push_back where appropriate